internal/pkg/client/oci/oci.go

```diff
@@ -114,17 +115,22 @@ func ImageSHA(ctx context.Context, uri string, sys *types.SystemContext) (string
 	return calculateRefHash(ctx, ref, sys)
 }
 
-func calculateRefHash(ctx context.Context, ref types.ImageReference, sys *types.SystemContext) (string, error) {
+func calculateRefHash(ctx context.Context, ref types.ImageReference, sys *types.SystemContext) (hash string, err error) {
 	source, err := ref.NewImageSource(ctx, sys)
 	if err != nil {
 		return "", err
 	}
+	defer func() {
+		if closeErr := source.Close(); closeErr != nil {
```

Comment:
I'm going to approve this because it's a cherry pick, but this construction is really strange.

In principle the idea with errors.Wrapf is that you have an error that you wish you annotate with additional information (like what was happening when the error occurred). Here you are annotating the original error err with an additional error. The wrapping should have been done on line ~ 120.

Also note that you have a variable error on the function's signature and on line 119 you have a short variable declaration, except that the only variable being declared is "source", not "err". This makes for hard-to-read code, as quickly glancing at the code might look like the anonymous function on line 123 is a closure on the variable from line 119.